### ISSUE TITLE
Automatically create test users for configured runmodes

### ISSUE THREAD

**Issue by ghenzler** [Netcentric/accesscontroltool#208]
It is often required for to have exactly one test user per role to be able to test, but only on staging environments. This could be automatically generated when providing the following configuration:

```
- global_config:
      autoCreateTestUsers:
           createForGroupNamesRegEx: "myproj.*" # to select the groups test users should be created for
           prefix: "tu-"  # prefix is required
           path: /home/users/myproject-test-users
           password: "mypw"   # a generic pw for the test users, if left out defaults to user name
           skipForRunmodes: prod, production # can be given to skip for certain runmodes (prod, production is the default)
```

**Comment by mittalkunal** [comment]
@ghenzler 

When is this feature likely to be available?

Thanks
Kunal

**Comment by ghenzler** [comment]
@mittalkunal I have not had time to work on it, feel free to create a PR

**Comment by ghenzler** [comment]
After having thought about this a bit more, it is probably easier/more flexible to extend the existing LOOP functionality with loops as follows:

```
- FOR group IN AUTHORIZABLES MATCHING PATH /home/groups/myproj/.*:
    - tu-${group.name}:
        - name: "Test User ${group.profileName}"
          isMemberOf: ${group.name}
          password: fixedPw  # could be set to tu-${group.name} as well
          path: ${replace(group.path,'/groups/','/users/testusers/')}

```

**Comment by soumyahazra** [comment]
@ghenzler In which version this feature is going to be available? Also, how can we avoid getting the test users created on the Production environment with this approach?

**Comment by ghenzler** [comment]
After trying to implement the `FOR group IN AUTHORIZABLES MATCHING PATH` approach there are two problems:
* If authorizables are taken from repo, they don't exist in the first run and hence the test users would only be created on second run
* If authorizables are taken from configuration, it is a hen and egg problem between the authorizables that are looped over and the authorizables that are added to the config at the same time.

Hence I would go back to the inital suggestion using a syntax like proposed in https://github.com/Netcentric/accesscontroltool/issues/208#issue-241178384

**Comment by ghenzler** [comment]
@soumyahazra @mittalkunal This is included in release [v2.1.0](https://github.com/Netcentric/accesscontroltool/releases/tag/2.1.0)
